[Edward Mualim] iP

Edfernape opened the pull request:
## Details of Duke
- **easy to use** chatbot
- keeps track of your tasks
> "So amazing! - ~~Not the author~~"

### Some Commands Available
1. `todo`
2. `deadline`

### Example Usage
```
todo Have a fruit
```
### An example of your task list
- [x] Have a fruit
- [ ] Drink some water
- [ ] Take a nap

### Totally irrelevant json code
```json
{
  "firstName": "Not",
  "lastName": "The Author",
  "age": 25
}
```

[Link to author's github](https://github.com/edfernape) :smile:

Edfernape committed:
Created the classes Chatbox and Task.

Edfernape committed:
Implemented Level-1.

Edfernape committed:
Implemented Level-2.

Edfernape committed:
Implement Level-3.

Edfernape committed:
Implement Level-4.

Edfernape committed:
Add A-TextUiTesting.

Edfernape committed:
Implement Level-5

Edfernape committed:
Implement Level-6.

johnbenedictyan commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,134 @@
+                        break;
+                    case KeywordChecker.EXACT_KEYWORD_LIST:
+                        duke.listTasks(duke);
+                        break;
```
Should there be an indentation for case clauses?

johnbenedictyan commented on src/main/java/KeywordChecker.java:
```diff
@@ -0,0 +1,87 @@
+        return hasTaskKeyword;
+    }
+
+    public static String getNonexactKeyword(String userInput) {
```
Perhaps `getNonExactKeyword` would be a better method name?

johnbenedictyan commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,134 @@
+                task -> { System.out.println(String.format("%d.%s",
+                        tasks.indexOf(task) + 1,
+                        task.printTask()));
+                });
```
Maybe this function could be optimised to make use of lambda functions.
The function `printf` also provides the functionality of both `println` and `String.format`.

```
interaction.tasks.forEach(
    task -> System.out.printf("%d.%s%n",
        tasks.indexOf(task) + 1,
        task.printTask()));
```

johnbenedictyan commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,18 @@
+                this.tag,
+                super.printTask(),
+                this.time);
+    }
```
Any reason why you created a custom method instead of overriding the task's `toString` method?

johnbenedictyan commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,18 @@
+public class Deadline extends Task {
+    private final char tag = 'D';
+    public static final String DELIMITER = " /by ";
+    private String time;
```
Perhaps this field could be final?

WuHaohui1231 reviewed (commented):
Nice handling of the exceptions!

WuHaohui1231 commented on src/main/java/KeywordChecker.java:
```diff
@@ -51,6 +54,8 @@ public static String getNonexactKeyword(String userInput) {
+            nonexactKeyword = KeywordChecker.MARK_KEYWORD_DELETE;
         } else if (userInput.startsWith(KeywordChecker.TASK_KEYWORD_TODO)) {
             nonexactKeyword = KeywordChecker.TASK_KEYWORD_TODO;
         } else if (userInput.startsWith(KeywordChecker.TASK_KEYWORD_DEADLINE)) {
```
Perhaps you can use a switch to avoid many else-ifs

WuHaohui1231 reviewed (commented):
Overall, I like your code!

Edfernape committed:
Add Level 7

Edfernape committed:
Fix updating data file bug

Edfernape committed:
Implement Level-8

Edfernape committed:
Add more OOP, packages and gradle

Edfernape committed:
Add junit tests

Edfernape committed:
Add JavaDocs

Edfernape committed:
Adhere to coding standard

Edfernape committed:
Implement the find feature

Edfernape committed:
Add checkstyle

Edfernape committed:
Add gui dependencies

Edfernape committed:
Add minimal UI

Edfernape committed:
Add proper logic to GUI

* Separated the UI logic into the Ui.java file for easier understanding.
* Added a process method to Duke to handle user input from GUI easier.
* Modified relevant methods in TaskList and Duke to return a String.

Doing so makes it easier for Ui to create a reply for the user to see.

Edfernape committed:
Add skeleton archive code.

Edfernape committed:
Add archive functionality.

Duke does not currently have a way to transfer or backup data.
Adding a keyword for the user to archive all recorded tasks would help
to solve this problem. Logic for this has been implemented in Parser,
Duke and Storage.

Edfernape committed:
Add Ui.png

Edfernape committed:
Add a user guide.

Edfernape committed:
Fix archive bug

Edfernape committed:
Add mock archive file

Edfernape committed:
Add acknowledgements in user guide's about section.

Edfernape committed:
Correct hyperlink in readme

Edfernape committed:
Add shadow

Edfernape committed:
Merge pull request #2 from Edfernape/branch-A-CodeQuality

Add shadow

Edfernape committed:
Remove Ui.png

Edfernape committed:
Re-add Ui.png

Edfernape committed:
Fix checkstyle errors

Edfernape committed:
Add shadowjar details in build.gradle

Edfernape committed:
Remove chimchar.PNG and pikachu.PNG

Edfernape committed:
Add images with correct lower case .png

Edfernape committed:
Update header comments to follow convention

Edfernape committed:
Add regex to better handle invalid user inputs
